dev-packages/application-package/src/extension-package-collector.ts

```diff
@@ -36,13 +36,20 @@ export class ExtensionPackageCollector {
     }
 
     protected collectPackages(pck: NodePackage): void {
-        if (!pck.dependencies) {
-            return;
+        if (pck.dependencies) {
```

Comment:
Suggestion: I believe we can cut down on the eslint-disable stuff as well as the duplicated for-loop by replacing the method body with something like this:

```ts
for (const [dependency, versionRange] of [
    ...Object.entries(pck.dependencies ?? {}),
    ...Object.entries(pck.peerDependencies ?? {})
]) {
    this.collectPackage(dependency, versionRange!);
}
```

Reply:
Thanks! That's a lot cleaner and works like charm. I have updated the PR accordingly.